src/celutil/indexlist.h

```diff
@@ -0,0 +1,81 @@
+// indexlist.h
+//
+// Copyright (C) 2023-present, the Celestia Development Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+
+#include <limits>
+#include <vector>
+
+namespace celestia::util
+{
+template <
+    typename T,
+    std::enable_if_t<std::is_integral_v<T>, bool> = true
+>
+constexpr T
+IndexListCapacity(T nSlices, T nPoints)
+{
+    return (nSlices * 2 + 4) * (nPoints - 1) - 2;
+}
+
+template <
+    typename T,
+    std::enable_if_t<std::is_integral_v<T>, bool> = true
+>
+void
+BuildIndexList(T I, T J, std::vector<T> &out)
+{
+    assert(I * J < std::numeric_limits<T>::max);
```

Comment:
I think this needs to be done as a division instead, and `max()` needs to be done as a method call:
```cpp
assert(std::numeric_limits<T>::max() / I >= J);
```

Reply:
Might be worth calling `reserve` on the vector before pushing the nodes?

Reply:
yes, it does nothing if a new capacity is less then the current, so it's safe to add it